canvas/text.go

```diff
@@ -21,6 +21,11 @@ type Text struct {
 	Text      string         // The string content of this Text
 	TextSize  float32        // Size of the text - if the Canvas scale is 1.0 this will be equivalent to point size
 	TextStyle fyne.TextStyle // The style of the text content
+
+	// FontSource defines a resource that can be used instead of the theme for looking up the font.
+	//
+	// Since: 2.5
+	FontSource fyne.Resource
```

Comment:
Is fyne.Resource becoming too overloaded? (it can be a font, an SVG, anything really) - Do we need a new type fyne.Font and an API to attempt to parse a fyne.Resource into a fyne.Font and return an error if it cannot?

Reply:
Yes it can literally be anything. It was never image specific but was mostly used there.
Perhaps we should look into the idea of specific types of resources like image/font etc... but there is no precedent at this time.